### PR TITLE
Fix issues with GitFileManager's nuget.config updater

### PR DESCRIPTION
**Pull request by MattGal**
… to work around existing comments, and never delete them. Update tests to reflect new expected behaviors

Context: https://github.com/dotnet/arcade/issues/6808

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/inputs/NugetConfigFiles/RemovingAllSourcesKeepsTheComments/NuGet.output.config [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+  </packageSources>
+  <disabledPackageSources>
+    <add key="dotnet-coreclr" value="true" />
+    <clear />
+    <add key="nuget.org" value="true" />
+    <clear />
+    <!--Begin: Package sources managed by Dependency Flow automation. Do not edit the sources below.-->
+    <!--  Begin: Package sources from dotnet-arcade -->
+    <add key="darc-int-dotnet-arcade-b043797" value="true" />
```
Comment (MattGal) [inline src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/inputs/NugetConfigFiles/RemovingAllSourcesKeepsTheComments/NuGet.output.config:32]: @mmitche given these feeds were removed, do they need to be removed from disabled?  Disabling a non-existent key is a no-op.  If I should walk them and delete them I can, it just doesn't change much.

File: src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs [hunk 2]
```diff
@@ -401,18 +401,13 @@ public XmlDocument UpdatePackageSources(XmlDocument nugetConfig, Dictionary<stri
...
                 else if (currentNode.NodeType == XmlNodeType.Comment)
                 {
-                    if (currentNode.Value.Equals(MaestroBeginComment, StringComparison.OrdinalIgnoreCase) ||
-                        currentNode.Value.Equals(MaestroEndComment, StringComparison.OrdinalIgnoreCase))
+                    if (currentNode.Value.Equals(MaestroBeginComment, StringComparison.OrdinalIgnoreCase))
                     {
-                        withinMaestroComments = currentNode.Value.Equals(MaestroBeginComment, StringComparison.OrdinalIgnoreCase);
-                        currentNode = RemoveCurrentNode(currentNode);
-                        continue;
+                        withinMaestroComments = true;
```
Comment (mmitche) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs:406]: What happens here if someone malforms the comment? e.g. they manually delete the end comment but not the beginning, or vice-versa.

Comment (MattGal) [reply src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs:406]: - if it's not valid XML, we never get this far
- If they delete either comment, the code the matching checks like this catch them:

```
            XmlComment startCommentBlock = GetFirstMatchingComment(disabledSourcesNode, MaestroBeginComment);
            if (startCommentBlock != null)
            {	            {
                var managedSources = GetManagedPackageSources(maestroManagedFeedsByRepo[repoName]).OrderByDescending(t => t.feed).ToList();	                insertAfterNode = startCommentBlock;
            }
```

I can easily add a test case or two for this scenario too.

File: src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/inputs/NugetConfigFiles/RemovingAllSourcesKeepsTheComments/NuGet.output.config (hunk 1 above)
Comment (mmitche) [reply src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/inputs/NugetConfigFiles/RemovingAllSourcesKeepsTheComments/NuGet.output.config:32]: Yes, they should be removed. If they aren't, then over time the number of disabled sources tends to get very large.

**Review by mmitche — APPROVED** [review]
Looks generally good. I would add a test case for a few things:
- If the overall begin/end comments are malformed (can find one but not the other).
- If the repo-specific begin/end comments are malformed (can find one but not the other).
- If the repo-specific comments are not properly alphabetized.
- (If it doesn't already exist) Substitution of some feeds and removal of others. E.g. Repo A and B and C exist in the package sources right now, and we can updates for A and B but C is removed.

For the cases where the current status is malformed, I would not necessarily expect the system to completely fix things up (unless that's really easy to do). Maybe it introduces a new begin/end block, etc. But I wouldn't expect it to continue to alter the config over time (e.g. continually add new comments or feeds).

So, you might have your test cases do the following:
1. Take input, apply managed feeds, check output.
2. Re-apply managed feeds, check that output is still the same.

**Comment by MattGal** [comment]
> Looks generally good. I would add a test case for a few things:
> 
> * If the overall begin/end comments are malformed (can find one but not the other).
> * If the repo-specific begin/end comments are malformed (can find one but not the other).
> * If the repo-specific comments are not properly alphabetized.
> * (If it doesn't already exist) Substitution of some feeds and removal of others. E.g. Repo A and B and C exist in the package sources right now, and we can updates for A and B but C is removed.
> 
> For the cases where the current status is malformed, I would not necessarily expect the system to completely fix things up (unless that's really easy to do). Maybe it introduces a new begin/end block, etc. But I wouldn't expect it to continue to alter the config over time (e.g. continually add new comments or feeds).
> 
> So, you might have your test cases do the following:
> 
> 1. Take input, apply managed feeds, check output.
> 2. Re-apply managed feeds, check that output is still the same.

Added the scenarios you called out, also every test has been running the process again on the output XML and ensuring idempotency since fixing the first bug I made :)

**Review by mmitche — APPROVED** [review]

**Pull request merged**